### ISSUE TITLE
WaitHandle.Dispose – diposing is not necessary

### ISSUE THREAD

**Issue by Uzivatel919** [dotnet/dotnet-api-docs#7507]
>  Note
>
> Always call Close or Dispose() before you release your last reference to the WaitHandle. Otherwise, the resources it is using will not be freed.

While fact that `WaitHandle` uses `SafeWaitHandle` that will be finalized when `Dispose` not called is implementation detail I guess this note better avoid direct lying. I cannot tell for previous version though.

**Comment by MSDN-WhiteKnight** [comment]
It is lying technically, but always calling Dispose is still proper recommendation. Finalizer in many cases might be not called, for example, because [object is rooted for longer then it should be](https://github.com/dotnet/docs/issues/24440) or [finalizer thread is stuck on some other deadlocked finalizer](https://github.com/dotnet/docs/issues/17463#issuecomment-600732942). Finalization is no longer recommended cleanup approach on CoreCLR, and docs should steer people towards explicitly disposing everything. Better keep this note unchanged.

**Comment by Uzivatel919** [comment]
Fine info. So what about

_Always call Close or Dispose() before you release your last reference to the WaitHandle. Otherwise, the resources it is using **might not be freed depending on actual implementation**._

Also “_before you release your last reference to the WaitHandle_” implies that `WaitHandle` is used _inline_. That seems to me rather uncommon in practice. Usually when `Dispose` is discussed, it is used something like “_Always call `Dispose` on objectXY when you are finished using it…”_.

For me it should be thus similar to 

_Whenever `WaitHandle` object is no longer needed call `Close` or `Dispose` method. Otherwise, the resources it is using might not be freed depending on actual implementation._

>  Close or Dispose() 

Also note inconsistency: `Close` has no parameter list and `Dispose` has one.

**Comment by MSDN-WhiteKnight** [comment]
>Usually when Dispose is discussed, it is used something like “Always call Dispose on objectXY when you are finished using it…”.

Yeah, it's usual boilerplate wording for notes on IDisposable types, and remarks of WaitHandle class itself already use it: https://docs.microsoft.com/en-us/dotnet/api/system.threading.waithandle?view=net-6.0#remarks

"This type implements the IDisposable interface. When you have finished using the type or a type derived from it, you should dispose of it either directly or indirectly..."

Seems fine to me if Dispose method's remarks would use similar wording for consistency.

>Also note inconsistency: Close has no parameter list and Dispose has one.

There's a different xref syntax for some reason: https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Threading/WaitHandle.xml#L198 - Close references method group, but Dispose references a concrete method without parameters. Not a big deal, but could also be fixed.

**Comment by Uzivatel919** [comment]
Same wording uses [WaitHandle.Close Method](https://docs.microsoft.com/en-us/dotnet/api/system.threading.waithandle.close?view=net-6.0#System_Threading_WaitHandle_Close).
